step-strikers-app/step-strikers-app/Classes/RPGCharacter.swift

```diff
@@ -0,0 +1,174 @@
+//
+//  RPGCharacter.swift
+//  step-strikers-app
+//
+//  Created by Jalyn Derry on 2/7/23.
+//
+
+class RPGCharacter {
+    var characterName: String
+    var userName: String
+    
+    // Character Stats Variables
+    var maxHealth: Int
+    var currHealth: Int
+    var maxStamina: Int
+    var currStamina: Int
+    
+    // Inventory Variables
+    var weaponsInInventory: [Weapon]
+    var currWeapon: Weapon
+    var armorInInventory: [Armor]
+    var currArmor: Armor
+    
+    // Rolling modifiers
+    var attackModifier = 0
+    var defenseModifier = 0
+    var magicResistanceModifier = 0
+    
+    init(characterName: String, userName: String, health: Int,
+         stamina: Int){
+        self.characterName = characterName
+        self.userName = userName
+        self.maxHealth = health
+        self.currHealth = health
+        self.maxStamina = stamina
+        self.currStamina = stamina
+        
+        // Initialize Starter Inventory
+        self.weaponsInInventory = [fistsWeapon]
+        self.currWeapon = fistsWeapon
+        self.armorInInventory = [noArmorArmor]
+        self.currArmor = noArmorArmor
+    }
+    
+    // Universal Functions
+    func wield(weaponObject: Weapon){
+        if(weaponObject.checkIfProficient(wearer: self)){
+            self.currWeapon = weaponObject
+        } else {
+            print("\(self.characterName) is not proficient with \(weaponObject.name).\n They will use this weapon at a disadvantage")
+            self.currWeapon = weaponObject
+        }
+    }
+
+    func wear(armorObject: Armor){
+        if(armorObject.checkIfSuited(potentialWearer: self)){
+            self.currArmor = armorObject
+        } else {
+            print("\(armorObject.name) is not properly suited for \(self.characterName). Their armor class will be at a disadvantage")
+        }
+    }
+
+    func fight(target: inout RPGCharacter){
+        
+        let damageDealt = calculateDamage(wielder: self, target: target, damage: self.currWeapon.damage)
+        
+        decreaseHealth(amtDamage: damageDealt)
+
+        decreaseStamina()
+    }
+
+    func decreaseStamina(){
```

Comment:
I would add a check to make sure it doesn't go below 0. 

currStamina = max(0, currStamina - weapon.staminacost)

You don't have to do this of course, I'm just thinking of cases where you have like -5 stamina. If you use some potion or something to increase your stamina by 5, then you'd still have 0. I think it would be better if you can't go below 0 (and that's what I've been doing in my spell code so far)

Reply:
Actually maybe disregard this. I would definitely do this for health but I don't think there's any way to get negative stamina

Reply:
So question about this:
I understand that if your hp reaches zero, you die but with low stamina what functionality occurs?

Reply:
I think nothing except you can only use your fists?